modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormGuestUploadFieldUtil.java

```diff
@@ -92,20 +91,20 @@ protected static boolean hasGuestUploadField(
 		Map<String, DDMFormField> ddmFormFieldsMap =
 			ddmForm.getDDMFormFieldsMap(true);
 
-		Optional<DDMFormField> ddmFormFieldOptional = Optional.empty();
+		boolean guestUploadField = false;
 
 		for (DDMFormField ddmFormField : ddmFormFieldsMap.values()) {
 			if (Objects.equals(ddmFormField.getType(), "document_library") &&
 				GetterUtil.getBoolean(
 					ddmFormField.getProperty("allowGuestUsers"))) {
 
-				ddmFormFieldOptional = Optional.of(ddmFormField);
+				guestUploadField = true;
 
 				break;
 			}
 		}
 
-		return ddmFormFieldOptional.isPresent();
+		return guestUploadField;
```

Comment:
@LisztLiszt Please simplify this like

```
		for (DDMFormField ddmFormField : ddmFormFieldsMap.values()) {
			if (Objects.equals(ddmFormField.getType(), "document_library") &&
				GetterUtil.getBoolean(
					ddmFormField.getProperty("allowGuestUsers"))) {

				return true;
			}
		}

		return false;
```

Please double check other changes in your PR. Thanks!